### DISCUSSION TITLE
Which nodes need GPU?

### DISCUSSION THREAD

**Discussion by douxiaofeng99** [milvus-io/milvus#33618] [Q&A and General discussion]
In the milvus architecture, which node need gpu and how much? the nodes are Query Nodes, Data Nodes, Index Node

**Comment by yhmo** [comment]
- Query node loads segments index into memory for search, including growing segments and sealed segments. If the index is GPU type, the query node needs GPU to compute.
- Data node persists data into segments, merges small segments into large segments.
- Index node builds indexes for sealed segments. If the index is GPU type, the index node needs GPU to build the index.
